libminifi/src/FlowController.cpp

```diff
@@ -473,8 +473,10 @@ std::vector<BackTrace> FlowController::getTraces() {
 
 std::map<std::string, std::unique_ptr<io::InputStream>> FlowController::getDebugInfo() {
   std::map<std::string, std::unique_ptr<io::InputStream>> debug_info;
-  if (auto logs = core::logging::LoggerConfiguration::getCompressedLog(true)) {
-    debug_info["minifi.log.gz"] = std::move(logs);
+  auto logs = core::logging::LoggerConfiguration::getCompressedLogs(true);
+  for (size_t i = 0; i < logs.size(); ++i) {
+    std::string index_str = i == logs.size() - 1 ? "" : "." + std::to_string(logs.size() - 1 - i);
+    debug_info["minifi.log" + index_str + ".gz"] = std::move(logs[i]);
```

Comment:
The logs can be decompressed inside the debug archive. The tarball itself is compressed anyway.

Reply:
I'm not sure if it's worth the performance impact the decompression may add to the agent. We introduced the compression because there may be a flood of repeated logs that may take up a lot of memory and this way we could highly reduce the stored content. If we decompress the logs here it can inflate exponentially which may cause a momentary resource usage surge, which can be unnecessary as we will compress this bundle again in the next step.